### PR TITLE
CloudServiceRP 2022-04-04

### PR DESCRIPTION
**Pull request by grizzlytheodore**
# Contributing to the Azure SDK

update with CloudService 2022-04-04 version 
swagger PR merged: https://github.com/Azure/azure-rest-api-specs/pull/19468

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by grizzlytheodore** [comment]
@Yao725 , please review

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/CloudServiceNetworkProfile.cs [hunk 1]
```diff
@@ -35,14 +35,17 @@ public CloudServiceNetworkProfile()
...
         /// Internal Load Balancer.</param>
+        /// <param name="slotType">Possible values include: 'Production',
+        /// 'Staging'</param>
         /// <param name="swappableCloudService">The id reference of the cloud
         /// service containing the target IP with which the subject cloud
         /// service can perform a swap. This property cannot be updated once it
         /// is set. The swappable cloud service referred by this id must be
         /// present otherwise an error will be thrown.</param>
-        public CloudServiceNetworkProfile(IList<LoadBalancerConfiguration> loadBalancerConfigurations = default(IList<LoadBalancerConfiguration>), SubResource swappableCloudService = default(SubResource))
+        public CloudServiceNetworkProfile(IList<LoadBalancerConfiguration> loadBalancerConfigurations = default(IList<LoadBalancerConfiguration>), string slotType = default(string), SubResource swappableCloudService = default(SubResource))
```
Comment (archerzz) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/CloudServiceNetworkProfile.cs:45]: Breaking change. We need a customization here.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/EncryptionSetIdentity.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public EncryptionSetIdentity()
         /// associated with the disk encryption set. The user identity
         /// dictionary key references will be ARM resource ids in the form:
         /// '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'.</param>
-        public EncryptionSetIdentity(string type = default(string), string principalId = default(string), string tenantId = default(string), IDictionary<string, EncryptionSetIdentityUserAssignedIdentitiesValue> userAssignedIdentities = default(IDictionary<string, EncryptionSetIdentityUserAssignedIdentitiesValue>))
+        public EncryptionSetIdentity(string type = default(string), string principalId = default(string), string tenantId = default(string), IDictionary<string, UserAssignedIdentitiesValue> userAssignedIdentities = default(IDictionary<string, UserAssignedIdentitiesValue>))
```
Comment (archerzz) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/EncryptionSetIdentity.cs:54]: It looks like `EncryptionSetIdentityUserAssignedIdentitiesValue` was renamed as `UserAssignedIdentitiesValue`. Can you confirm?

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineIdentity.cs [hunk 3]
```diff
@@ -47,7 +47,7 @@ public VirtualMachineIdentity()
         /// associated with the Virtual Machine. The user identity dictionary
         /// key references will be ARM resource ids in the form:
         /// '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'.</param>
-        public VirtualMachineIdentity(string principalId = default(string), string tenantId = default(string), ResourceIdentityType? type = default(ResourceIdentityType?), IDictionary<string, VirtualMachineIdentityUserAssignedIdentitiesValue> userAssignedIdentities = default(IDictionary<string, VirtualMachineIdentityUserAssignedIdentitiesValue>))
+        public VirtualMachineIdentity(string principalId = default(string), string tenantId = default(string), ResourceIdentityType? type = default(ResourceIdentityType?), IDictionary<string, UserAssignedIdentitiesValue> userAssignedIdentities = default(IDictionary<string, UserAssignedIdentitiesValue>))
```
Comment (archerzz) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineIdentity.cs:50]: rename `VirtualMachineIdentityUserAssignedIdentitiesValue` as `UserAssignedIdentitiesValue`?

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/CloudServiceExtensionProperties.cs [hunk 4]
```diff
@@ -132,14 +132,14 @@ public CloudServiceExtensionProperties(string publisher = default(string), strin
         /// Extension (like RDP), this is the XML setting for the extension.
         /// </summary>
         [JsonProperty(PropertyName = "settings")]
-        public string Settings { get; set; }
+        public object Settings { get; set; }
```
Comment (archerzz) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/CloudServiceExtensionProperties.cs:135]: This seems a breaking change. But looks like there is no customization to fix the backward compatibility.

**Review by archerzz — CHANGES_REQUESTED** [review]
@grizzlytheodore Can you add the customization for `CloudServiceNetworkProfile`? For other breaking changes, I could not think of an easy way around. If you could find a way, feel free to add it. Otherwise, leave it alone. Thanks.

**Review by Yao725 — COMMENTED** [review]
Why don't I see any update about compute_resource-manager?

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineIdentity.cs (hunk 3 above)
Comment (grizzlytheodore) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineIdentity.cs:50]: confirmed

**Comment by grizzlytheodore** [comment]
> Why don't I see any update about compute_resource-manager?

@Yao725  what do you mean? this is for CloudService version release which is part of Compute service. I believe some CRP related changes in this PR were some things left out from last changes to swagger

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/CloudServiceNetworkProfile.cs (hunk 1 above)
Comment (grizzlytheodore) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/CloudServiceNetworkProfile.cs:45]: done

**Comment by grizzlytheodore** [comment]
@Yao725  @archerzz  your comments are addressed. please take a look

**Comment by Yao725** [comment]
> > Why don't I see any update about compute_resource-manager?
> 
> @Yao725 what do you mean? this is for CloudService version release which is part of Compute service. I believe some CRP related changes in this PR were some things left out from last changes to swagger
@grizzlytheodore , every time when we try to update the sdk, [this file](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/mgmt/mgmtmetadata/compute_resource-manager.txt) should also be updated. This file will help us determine the specific commit in swagger repo and trigger some pipelines. Currently the commit points to the `tag: package-2022-03-02` and should be updated, so I'd like to know why this PR doesn't contain this change after you run the script generate.ps1.

**Comment by grizzlytheodore** [comment]
@Yao725 oh i see. I've updated it. I missed that step. We usually do it manually. Do you have any documentation on using script generate.ps1. and what it does?

**Comment by Yao725** [comment]
> @Yao725 oh i see. I've updated it. I missed that step. We usually do it manually. Do you have any documentation on using script generate.ps1. and what it does?

We use this script to call the `Start-AutoRestCodeGeneration` to generate the code, in this script you can specify the resource provider, code generation directory, tag, etc. Here is our [doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=track-1-sdk-generation-from-updated-spec) about how to update track 1 sdk.

**Review by archerzz — APPROVED** [review]

**Comment by archerzz** [comment]
@Yao725 Do you have any further comment?

**Comment by Yao725** [comment]
> 

No, approved and merged, thank you!

**Review by Yao725 — APPROVED** [review]

**Pull request merged**